app/src/main/java/com/hedvig/app/feature/keygear/data/DeviceInformationService.kt

```diff
@@ -0,0 +1,69 @@
+package com.hedvig.app.feature.keygear.data
+
+import android.annotation.SuppressLint
+import android.app.UiModeManager
+import android.content.Context
+import android.content.res.Configuration
+import android.provider.Settings
+import android.util.DisplayMetrics
+import android.view.WindowManager
+import com.hedvig.android.owldroid.type.KeyGearItemCategory
+import java.math.BigInteger
+import java.security.MessageDigest
+import kotlin.math.pow
+import kotlin.math.sqrt
+
+class DeviceInformationService(
+    private val context: Context
+) {
+    @SuppressLint("HardwareIds")
+    fun getDeviceFingerprint(): String {
+        val messageDigest = MessageDigest.getInstance("SHA-256")
+        messageDigest.reset()
+
+        val androidId = Settings.Secure.getString(context.contentResolver, Settings.Secure.ANDROID_ID)
+        messageDigest.update(androidId.toByteArray())
+
+        val result = messageDigest.digest()
+        return BigInteger(1, result).toString(16).padStart(32, '0')
```

Comment:
why do we do this?

Reply:
This is the best way I found to convert it from a byte array into a hex string 🙂